### ISSUE TITLE
Torch doesn't do initial validation phase

### ISSUE THREAD

**Issue by lukeyeager** [NVIDIA/DIGITS#120]
Torch takes forever to actually output something on the loss/accuracy graph, and when something finally appears, it doesn't start at x=0. The way to fix this is to run a validation pass *before* beginning training.

Caffe by default does a pass through the validation data once before beginning to train the model ([see comments here](https://github.com/NVIDIA/caffe/blob/v0.11.0/src/caffe/proto/caffe.proto#L134-L136)). They do this partly to ensure memory availability (which shouldn't be an issue with Torch), but it also helps DIGITS to be able to display something quickly.

**Comment by Sravan2j** [comment]
@lukeyeager: fixed the issue. Please test it once and close issue if fix looks good. Thanks for your time.

**Comment by lukeyeager** [comment]
That'll work. It doesn't have the initial training loss value, but that doesn't seem necessary.

Thanks!
